backend/cfg/cfg.ml

```diff
@@ -32,62 +32,209 @@ include Cfg_intf.S
 module BasicInstructionList = struct
   type instr = basic instruction
 
-  type cell =
+  type node =
     { instr : instr;
-      mutable before_rev : instr list;
-      mutable after : instr list
+      mutable prev : node;
+      mutable next : node
     }
 
-  let insert_before cell instr = cell.before_rev <- instr :: cell.before_rev
-
-  let insert_after cell instr = cell.after <- instr :: cell.after
-
-  let instr cell = cell.instr
-
-  type t = instr list ref
-
-  let make_empty () = ref []
-
-  let make_single instr = ref [instr]
-
-  let of_list l = ref l
-
-  let hd t = match !t with [] -> None | hd :: _ -> Some hd
+  (* CR xclerc for xclerc: a dummy instruction value has probably been
+     introduced by another pull request. *)
+  let dummy_instruction : instr =
```

Comment:
Can we avoid `dummy_*` by using a variant type like Queue.cell in the standard library?

Reply:
I was originally afraid that would made
some operations slower (because we
have to actually match while an invariant
tells us the nature of a value), but it is
possible that fear was not founded given
the set of operations currently implemented.